### PR TITLE
Change for handling state of vdc

### PR DESCRIPTION
**Pull request by vbauzysvmware**
ref: https://github.com/terraform-providers/terraform-provider-vcd/pull/380

Changes needed to handle system admin and org admin diff set data provided from API.

This allows to know when provisioning isn't returned.

### CODE REVIEW

**Comment by vbauzysvmware** [comment]
> Thanks for it. The only question is - are these fields enough or could there be more bools needed to convert to pointers.
> Also - a small note in changelog.

Found one more.

File: types/v56/types.go [hunk 1]
```diff
@@ -398,11 +398,11 @@ type AdminVdc struct {
...
 	ResourceGuaranteedMemory *float64   `xml:"ResourceGuaranteedMemory,omitempty"`
 	ResourceGuaranteedCpu    *float64   `xml:"ResourceGuaranteedCpu,omitempty"`
-	VCpuInMhz                int64      `xml:"VCpuInMhz,omitempty"`
-	IsThinProvision          bool       `xml:"IsThinProvision,omitempty"`
+	VCpuInMhz                *int64     `xml:"VCpuInMhz,omitempty"`
+	IsThinProvision          *bool      `xml:"IsThinProvision,omitempty"`
 	NetworkPoolReference     *Reference `xml:"NetworkPoolReference,omitempty"`
 	ProviderVdcReference     *Reference `xml:"ProviderVdcReference"`
-	UsesFastProvisioning     bool       `xml:"UsesFastProvisioning,omitempty"`
+	UsesFastProvisioning     *bool      `xml:"UsesFastProvisioning,omitempty"`
```
Comment (dataclouder) [inline types/v56/types.go]: Why do we have two pointers to Boolean and two straight Booleans in the same structure?
They are represented in the same way in XML.

Comment (vbauzysvmware) [reply types/v56/types.go]: This is requeste in ref: terraform-providers/terraform-provider-vcd#380

Issue in terraform that in state file we should know if it is false or wasn't provided

Comment (dataclouder) [reply types/v56/types.go]: This doesn't answer the question. Why two pointers and two straight ones?

Comment (vbauzysvmware) [reply types/v56/types.go]: With pointers we know that won't be returned from API when called by org admin

Comment (dataclouder) [reply types/v56/types.go]: I know what pointers are used for. 
My question is why we we two Booleans and two Boolean pointers in the same structure.
Why not all pointers?

Comment (vbauzysvmware) [reply types/v56/types.go]: I mean. I made pointers which I know API will return or not according role, on others I don't have info

Comment (dataclouder) [reply types/v56/types.go]: Thanks.
FYI: the field `OverCommitAllowed` is ignored both in creation and in update.

**Comment by vbauzysvmware** [comment]
> LGTM, although I think that we should be more consistent with the fields in VDC data structure.
> We should treat all Boolean fields in the same way.

Thank you for review. P.s. it's not just booleans. Also I feel that we may come back later to this, that all values should be pointers to know if they were provided by API or not.

**Comment by vbauzysvmware** [comment]
> Request and a question:
> 
>     * Change log missing
> 
>     * Why we are not adding add pointers to OverCommitAllowed and VmDiscoveryEnabled too?

* changelog added
* these two values aren't used and seen xml

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 2.5.0 
+
+* Change fields ResourceGuaranteedCpu, VCpuInMhz, IsThinProvision, NetworkPoolReference,ProviderVdcReference and UsesFastProvisioning in AdminVdc to pointers
```
Comment (lvirbalas) [inline CHANGELOG.md]: Please add also why this change is needed. Also, there's a space missing after one comma.

Comment (vbauzysvmware) [reply CHANGELOG.md]: done
